### DISCUSSION TITLE
Could you keep the public cloud service?

### DISCUSSION THREAD

**Discussion by lihugang** [deta/space-docs#15] [Ideas]
I have tried Deta Space and I have known it's a personal cloud.  
I'm hosting the official website and web demo for my open source projects through Deta Cloud. But after Deta Cloud stopped serving on May 1, users will not be able to access these contents. They will definitely not sign up for Deta Space to view the official website and web demos, so can you keep the service of Deta Cloud or allow me to create a public cloud in Deta Space to host my project. I found that Builder Instance seems to have a similar function,  ~~so I would like to ask about the life cycle of Builder Instance and whether it is possible to configure a custom `xxx.deta.app` domain name for Instance in the future~~. I have read the <https://deta.space/manual/features/custom-domains> and I plan to register a domain and bind it to Deta App Instance
I really appreciate you viewing my suggestions.

**Comment by CyberL1** [comment]
You can use the `public_routes` in Spacefile to make the entire of your app's content avaliable to everyone without the need to register https://deta.space/docs/en/basics/micros#public-routes

**Reply by lihugang** [reply]
I appreciate your answer. I’ve migrated my websites from cloud to space with 
```yaml
public_routes:
  - *
``` 
and setting a custom domain.

**Reply by CyberL1** [reply]
> I appreciate your answer. I’ve migrated my websites from cloud to space with `public_routes: *` and setting a custom domain.

No `public_routes` is an array of paths you want to be public, not a single string
![obraz](https://user-images.githubusercontent.com/44707958/219855097-14c6d79e-d77d-497b-8557-8c0c468ab08a.png)

So, if you want all of your app to be public, it'll be something like:
````yml
public_routes:
  - "/*"
 ````

**Reply by lihugang** [reply]
sorry, when I replied your suggestion, I didn't view my code and I wrote a value randomly. Thanks for your answer.

**Reply by s4m-mo** [reply]
When I used public routes, I made my entire app public, and there are no bugs in my code, it works on my end, but it comes up with the following error when I try to request any route.

```Error Type:
Runtime.ImportModuleError

Error Message:
Unable to import module '_entry': No module named '_entry'

Logs:
time="2023-02-20T14:49:56Z" level=info msg="serving logs listener on sandbox.localdomain:1234" agent=logsApiAgent
TELEMETRY	Name: telemetry-extension	State: Subscribed	Types: [Function]
```

What could be the problem? Also, what port should my REST API be listening to?

**Reply by Lin-Rexter** [reply]
> When I used public routes, I made my entire app public, and there are no bugs in my code, it works on my end, but it comes up with the following error when I try to request any route.
> 
> ```
> Runtime.ImportModuleError
> 
> Error Message:
> Unable to import module '_entry': No module named '_entry'
> 
> Logs:
> time="2023-02-20T14:49:56Z" level=info msg="serving logs listener on sandbox.localdomain:1234" agent=logsApiAgent
> TELEMETRY	Name: telemetry-extension	State: Subscribed	Types: [Function]
> ```
> 
> What could be the problem? Also, what port should my REST API be listening to?

me too

**Reply by sudo-subg** [reply]
hey @sam-the-programmer @Lin-Rexter, you guys need to name your entry point file to `main.py` like instructed [here](https://deta.space/docs/en/quickstart-guides/python) for python runtime.

> what port should my REST API be listening to?

The port is specified in the `PORT` env var of micros and you should configure your app to read from that so your app doesn't break if we change the port in future.
